test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -336,27 +317,207 @@ private void CompareWorkspaces(WorkspaceComparisonData a, WorkspaceComparisonDat
                     continue;
                 }
             }
+        }
+        /// <summary>
+        /// saves the workspace comparison object to disk in a .data file
+        /// </summary>
+        /// <param name="wcd1"></param>
+        /// <param name="filePathBase"></param>
+        /// <param name="executionDuration"></param>
+        /// <param name="modelGuidToIDMap"></param>
+        public static void SaveWorkspaceComparisonData(serializationTestUtils.WorkspaceComparisonData wcd1,
+            string filePathBase,
+            TimeSpan executionDuration,
+            Dictionary<Guid,string> modelGuidToIDMap = null)
+        {
+            var nodeData = new Dictionary<string, Dictionary<string, object>>();
+            foreach (var d in wcd1.NodeDataMap)
+            {
+                var t = wcd1.NodeTypeMap[d.Key];
+                var nodeDataDict = new Dictionary<string, object>();
+                nodeDataDict.Add("nodeType", t.ToString());
+                nodeDataDict.Add("portValues", d.Value);
+                nodeData.Add(d.Key.ToString(), nodeDataDict);
+            }
 
-            foreach (var kvp in a.InputsMap)
+            var workspaceDataDict = new Dictionary<string, object>();
+            workspaceDataDict.Add("nodeData", nodeData);
+            workspaceDataDict.Add("executionDuration", executionDuration.TotalSeconds);
+
+            var dataMapStr = JsonConvert.SerializeObject(workspaceDataDict,
+                            new JsonSerializerSettings()
+                            {
+                                Formatting = Formatting.Indented,
+                                ReferenceLoopHandling = ReferenceLoopHandling.Ignore
+                            });
+
+            var dataPath = filePathBase + ".data";
+            if (File.Exists(dataPath))
             {
-                var vala = kvp.Value;
-                var valb = b.InputsMap[kvp.Key];
-                Assert.AreEqual(vala, valb, "input datas are not the same.");
+                File.Delete(dataPath);
```

Comment:
try..catch ?

Reply:
can do.

Reply:
hmm, actually now that I think about it a bit more, I don't see why I would want to catch this, all I will do is rethrow it. - if we fail to delete or save the file, I want the test to fail.